src/system_explorer_app.jl

```diff
@@ -515,20 +591,101 @@ callback!(
         c_type = getproperty(PowerSystems, Symbol(component_type))
         component = get_component(c_type, get_system(), c_name)
         ta = get_time_series_array(ts_type, component, ts_name)
-        trace = scatter(;
+        trace = PlotlyJS.scatter(;
             x = TimeSeries.timestamp(ta),
             y = TimeSeries.values(ta),
             mode = "lines+markers",
             name = c_name,
         )
         push!(traces, trace)
     end
-    layout = Layout(;
+    layout = PlotlyJS.Layout(;
         title = "$component_type SingleTimeSeries",
         xaxis_title = "Time",
         yaxis_title = "val",
     )
-    return plot([x for x in traces], layout)
+    return PlotlyJS.plot([x for x in traces], layout)
+end
+
+function plotlyjs_syncplot(plt::Plots.Plot{Plots.PlotlyJSBackend})
+    plt[:overwrite_figure] && Plots.closeall()
+    plt.o = PlotlyJS.plot()
+    traces = PlotlyJS.GenericTrace[]
+    for series_dict in Plots.plotly_series(plt)
+        filter!(p -> !(isa(last(p), Number) && isnan(last(p))), series_dict)
+        plotly_type = pop!(series_dict, :type)
+        series_dict[:transpose] = false
+        push!(traces, PlotlyJS.GenericTrace(plotly_type; series_dict...))
+    end
+    PlotlyJS.addtraces!(plt.o, traces...)
+    layout = Dict([
+        p for p in Plots.plotly_layout(plt) if first(p) ∉ [:xaxis, :yaxis, :height, :width]
+    ])
+    layout[:xaxis_visible] = false
+    layout[:yaxis_visible] = false
+    PlotlyJS.relayout!(plt.o, layout)
+    return plt.o
+end
+
+callback!(
+    app,
+    Output("map_plot", "figure"),
+    Input("plot_map_button", "n_clicks"),
+    Input("load_shp_button", "n_clicks"),
+    State("shp_text", "value"),
+) do n_clicks, n_shp_clicks, shp_txt
+    n_clicks < 1 && throw(PreventUpdate())
+
+    if n_shp_clicks > 0
+        shp_path = shp_txt
+    else
+        shp_path = joinpath(
+            pkgdir(PowerSystemsApps),
+            "src",
+            "assets",
+            "world-administrative-boundaries",
+            "world-administrative-boundaries.shp",
+        )
+    end
+
+    if endswith(shp_path, ".shp")
+        # load a shapefile
+        shp = PowerSystemsMaps.Shapefile.shapes(PowerSystemsMaps.Shapefile.Table(shp_path))
+        shp = PowerSystemsMaps.lonlat_to_webmercator(shp) #adjust coordinates
+
+        # plot a map from shapefile
+        p = Plots.plot(
+            shp,
+            fillcolor = "grey",
+            background_color = "#1E1E1E",
+            linecolor = "darkgrey",
+            axis = nothing,
+            grid = false,
+            border = :none,
+            label = "",
+            legend_font_color = :red,
+        )
+    else
+        p = Plots.plot(background_color = "black", axis = nothing, border = :none)
+    end
+
+    if !isnothing(get_system())
```

Comment:
Do you want to check for this condition on the first line of function (along with n_clicks) and then return early?

Reply:
no, because right now it will plot your shapefile without the network, which seems useful for checking the shapefile display.